lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -4,43 +4,60 @@ import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
 export class ConfigMigrationCommitMessageFactory {
-  private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
+  constructor(
+    private readonly config: RenovateConfig,
+    private readonly configFile: string
+  ) {}
 
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
-  }
-
-  create(): CommitMessage {
+  private create(messageTopic: string): CommitMessage {
     const { commitMessage } = this.config;
+    let { commitMessageAction, commitMessageTopic } = this.config;
 
-    this.config.commitMessageAction =
-      this.config.commitMessageAction === 'Update'
-        ? ''
-        : this.config.commitMessageAction;
+    if (commitMessageAction === 'Update') {
+      commitMessageAction = '';
+    }
 
-    this.config.commitMessageTopic =
-      this.config.commitMessageTopic === 'dependency {{depName}}'
-        ? `Migrate config ${this.configFile}`
-        : this.config.commitMessageTopic;
+    if (commitMessageTopic === 'dependency {{depName}}') {
```

Comment:
Why/how would this occur? I don't know why we're even looking at this logic here as it's only for the config migration PR and that has a fairly fixed title

Reply:
That's exactly [what I though](https://github.com/renovatebot/renovate/pull/16781#discussion_r933934162) when revisited this code, i dont think any of these should be user configurable.

`'dependency {{depName}}'` & `'Update'` are the defaults

Reply:
`commitMessageFactory` is what creates the semantic commit prefix. 
the only thing that we might still want to be user configurable is 
`semanticCommitType`. if we hard code it than `commitMessageFactory` is indeed redundant.

Reply:
now only `semanticCommitType` & `commitMessagePrefix/Suffix` are configurable.